app/models/userModel.py

```diff
@@ -70,6 +75,54 @@ def register(email, password, firstname, lastname):
             print(str(e))
             return None
 
+    @staticmethod
+    def update_email(email, id):
+        try:
+            rows = app.db.execute("""
+UPDATE Users
+SET email = :email
+WHERE id = :id
+""",
+                                  email=email, id = id)
+            return UserModel.get(id)
+        except Exception as e:
+            # likely email already in use; better error checking and reporting needed;
+            # the following simply prints the error to the console:
+            print(str(e))
+            return None
+
+    @staticmethod
+    def update_firstname(firstname, id):
+        try:
+            rows = app.db.execute("""
+UPDATE Users
+SET firstname = :firstname
+WHERE id = :id
+""",
+                                  id=id, firstname = firstname)
+            return UserModel.get(id)
+        except Exception as e:
+            # likely email already in use; better error checking and reporting needed;
```

Comment:
Error commenting here could be more specific

Reply:
Yep sounds good, I'll edit these along with the validations